### DISCUSSION TITLE
JSON at the device level

### DISCUSSION THREAD

**Discussion by obbers** [merbanan/rtl_433#2192] [Q&A] [answered]
Sorry if this is the wrong place but I'm trying to get the last entry into JSON rather than individual topics.

So ultimately my topic would look like this rtl433/devices/LaCrosse-BreezePro/751341 and it would contain a json payload of temp. frequency. etc rather than topics for each of those.

I'm using the rtl_433.conf instead of a command line, so the only output I have is
output mqtt://x.x.x.x:1883,retain=1

Any help would be appreciated.

**Comment by zuckschwerdt** [comment]
That mqtt output should just work. The topic would be `rtl_433/HOSTNAME/events` with a content of e.g. `{"time":...,"model":...","temperature_C":...}`

**Comment by obbers** [comment]
That is correct.  The events topic is json.  What I'm looking for is to
have JSON in the /devices topic,

**Comment by zuckschwerdt** [comment, marked as answer]
Ok, so you want an expanded topic for events? You can use the topic options for that:
`"mqtt://localhost:1883,events=rtl433/devices[/type][/model][/subtype][/channel][/id]"`
and you can also combine them by adding
`,devices=rtl433/devices[/type][/model][/subtype][/channel][/id]`
(note that "devices" uses further subtopics i.e. there is no collision.

**Reply by obbers** [reply]
That's exactly what I was looking for.  Thanks so much.
